### PR TITLE
Force rebuild of OVS/OVN sources during cluster build

### PR DESCRIPTION
**Pull request by markdgray**
When running "ovn_cluster.sh build", changes in OVN or OVS
source files are not detected. This patch deletes the
temporary "ovs" and "ovn" directory if present.

Signed-off-by: Mark Gray <mark.d.gray@redhat.com>

### CODE REVIEW

File: ovn_cluster.sh
Comment (flavio-fernandes) [inline ovn_cluster.sh:719]: NIT: let's move this echo to line 724

File: ovn_cluster.sh
Comment (flavio-fernandes) [inline ovn_cluster.sh:725]: NIT: let's move this echo to line 730

File: ovn_cluster.sh
Comment (flavio-fernandes) [inline ovn_cluster.sh:731]: rm -rf can be safely done if the file is there, let's remove this conditional

File: ovn_cluster.sh
Comment (flavio-fernandes) [inline ovn_cluster.sh:735]: same as comment above. Maybe combine these as:
```rm -rf ovs ovn```

**Review by flavio-fernandes — CHANGES_REQUESTED** [review]
Interesting request. I'm okay with it, but it changes a little bit the existing behavior on
explicitly re-using the src/ovn src/ovs directories if they were present. I'm guessing you don't want that, right?

Do you see a situation when that could be wanted and if so, have a new knob added that would honor the existing src directories?

Another way to approach this is to simply fail if the destination directories are present and force the user to remove them and retry. That would be safer but potentially more annoying. wdyt?

File: ovn_cluster.sh
Comment (markdgray) [reply ovn_cluster.sh:731]: > Interesting request. I'm okay with it, but it changes a little bit the existing behavior on
> explicitly re-using the src/ovn src/ovs directories if they were present. I'm guessing you don't want that, right?
> 
> Do you see a situation when that could be wanted and if so, have a new knob added that would honor the existing src directories?
> 
> Another way to approach this is to simply fail if the destination directories are present and force the user to remove them and retry. That would be safer but potentially more annoying. wdyt?

The use case that I have is that I have made a change to OVN code and I was expecting to rebuild based on this new code. However, unexpectedly - to me anyway, it was not rebuilding.

I think the fail would be more annoying but that is just for my use case.

I suppose the disadvantage of recreating the directories each time is performance?

Comment (flavio-fernandes) [reply ovn_cluster.sh:731]: > > Interesting request. I'm okay with it, but it changes a little bit the existing behavior on
> > explicitly re-using the src/ovn src/ovs directories if they were present. I'm guessing you don't want that, right?
> > Do you see a situation when that could be wanted and if so, have a new knob added that would honor the existing src directories?
> > Another way to approach this is to simply fail if the destination directories are present and force the user to remove them and retry. That would be safer but potentially more annoying. wdyt?
> 
> The use case that I have is that I have made a change to OVN code and I was expecting to rebuild based on this new code. However, unexpectedly - to me anyway, it was not rebuilding.
> 
> I think the fail would be more annoying but that is just for my use case.
> 
> I suppose the disadvantage of recreating the directories each time is performance?

nah, not an issue. I'm okay with merging the changes after making the changes we talked about.

**Review by flavio-fernandes — CHANGES_REQUESTED** [review]

File: ovn_cluster.sh
Comment (markdgray) [reply ovn_cluster.sh:731]: I reworked this as a compromise - should cover both use cases.

**Comment by numansiddique** [comment]
@markdgray Sorry that this PR is unattended for a while.

@flavio-fernandes  Is this PR good to go ? or something is pending ?

**Comment by flavio-fernandes** [comment]
> @markdgray Sorry that this PR is unattended for a while.
> 
> @flavio-fernandes Is this PR good to go ? or something is pending ?

Sorry for that. I'll check. Thanks for the bump. ;)

**Comment by markdgray** [comment]
@flavio-fernandes @numansiddique IIRC, there was something I wasn't happy with. Maybe hold off on this?

**Comment by flavio-fernandes** [comment]
> @flavio-fernandes @numansiddique IIRC, there was something I wasn't happy with. Maybe hold off on this?

Right. I guess that this pr is about a change in how we expect "build-images-with-ovn-rpms" to behave.

https://gist.github.com/flavio-fernandes/9e642823f8b6948bc4b12a2ef0b115ab

I am okay with this change if you are as well, @numansiddique @dceara

**Review by flavio-fernandes — APPROVED** [review]

File: ovn_cluster.sh
Comment (flavio-fernandes) [inline ovn_cluster.sh:740]: hm.... I was thinking about this some more.
If we are to do this, maybe we should get rid of these lines in Vagrantfile:
https://github.com/ovn-org/ovn-fake-multinode/blob/master/Vagrantfile#L45-L49
Could you please take care of that in this commit as well, please?

**Review by flavio-fernandes — CHANGES_REQUESTED** [review]
one more request... sorry.

**Comment by flavio-fernandes** [comment]
Hmm... no activity since Dec/20. Should we just "nevermind" this PR @markdgray ?

**Comment by markdgray** [comment]
Yes, I will close it

**Pull request closed without merging**
